### PR TITLE
Update upload UX

### PR DESCRIPTION
**Pull request by glacials**
Updates the visuals of uploading as well as offloads the parsing into an asynchronous job. Allows the run page to be completely loaded before parsing has finished, and shows a spinner until it completes, notified via Action Cable.

Multiuploads work better, and wait for all runs' parses to finish before refreshing, using Action Cable as well.

Fixed the blue upload call-to-action button that only shows for logged-in users with no runs.

This work sets us up to take parsing out of the web application and into a separate, independently scalable service like Lambda or a worker Rails box. That world will fix the issue of a user uploading a run so complex that it crashes or temporarily disables a web box. This branch doesn't get us completely there, but should help nonetheless as the worker is in its own Docker container pseudo-separate from the web service.

Somewhere during development of this branch I broke the run convert JS. I know we'd talked about whether we want to delete converts due to the extra maintenance work for something users can do manually with an upload -> export -> delete, so rather than debug it I killed the UI for it. If you think that's a bad move I can add it back in and debug.

### CODE REVIEW

File: app/jobs/parse_run_job.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class ParseRunJob < ApplicationJob
```
[the remark this answers is not in the document]
Comment (glacials) [reply app/jobs/parse_run_job.rb]: I actually see this as independent of API version; more of a standard internal way to parse a run asynchronously. I see this eventually absorbing `Run#parse_into_db`. It happens to notify a v4 channel when it's done, but unlike the race update jobs that is not its purpose. We can add more API version/channel calls to it if the time comes.

(Almost put the Action Cable broadcasts in `Run#parse_into_db`, but figured if we're parsing synchronously there'd be no one to hear the broadcasts)

File: Makefile [hunk 2]
```diff
@@ -59,3 +59,9 @@ clean:
 	rm -rf bundle/
 	rm -rf node_modules/
 	rm -f tmp/seed
+
+superclean:
```
[the remark this answers is not in the document]
Comment (BatedUrGonnaDie) [reply Makefile]: Does this remove the volumes as well?  In my experience I had to do `docker down --volumes` to get them as well.  Also we should probably remove the bundle and node modules folder as well.

File: app/javascript/channels/consumer.js [hunk 3]
```diff
@@ -2,9 +2,10 @@
 // You can generate new channels where WebSocket features live using the `rails generate channel` command.
 
 import { createConsumer } from "@rails/actioncable"
-
 import { getAccessToken } from '../token'
 
+const consumer = createConsumer(getWebsocketURL)
```
[the remark this answers is not in the document]
Comment (BatedUrGonnaDie) [reply app/javascript/channels/consumer.js]: Yeah, it was nice separating them but 2 connections is not good.  I've heard of tree shaking in the past where I believe it only takes one version of the dependency for your whole set of JS, that might be something to look into?

[the remark this answers is not in the document]
Comment (glacials) [reply app/javascript/channels/consumer.js]: Hmm looked briefly into it and it seems like tree shaking is a way of excluding orphaned modules from the pack, not necessarily of reusing one module's execution in multiple packs. Either way I don't want to block the branch on it, but let me know if I'm looking at the wrong thing.

[the remark this answers is not in the document]
Comment (jhobz) [reply app/javascript/channels/consumer.js]: Tree shaking is more of a developer tool intended to detect unused code and report on it for removal (or automatically remove it, sometimes even on import, as you're describing). Its implementations and coverage of various languages and codebases are hit or miss, depending on the specific frameworks involved.

**Pull request merged**